Content.Server/Chemistry/ReactionEffects/AreaReactionEffect.cs

```diff
@@ -0,0 +1,132 @@
+            serializer.DataField(ref _maxRange, "maxRange", 10);
+            serializer.DataField(ref _diluteReagents, "diluteReagents", false);
+            serializer.DataField(ref _reagentDilutionStart, "reagentDilutionStart", 4);
+            serializer.DataField(ref _reagentDilutionFactor, "reagentDilutionFactor", 1f);
+            serializer.DataField(ref _reagentMaxConcentrationFactor, "reagentMaxConcentrationFactor",2f);
+            serializer.DataField(ref _duration, "duration", 10f);
+            serializer.DataField(ref _spreadDelay, "spreadDelay", 0.5f);
+            serializer.DataField(ref _removeDelay, "removeDelay", 0.5f);
+            serializer.DataField(ref _sound, "sound", null);
+            serializer.DataField(ref _prototypeId, "prototypeId", null);
+        }
+
+        public void React(IEntity solutionEntity, double intensity)
+        {
+            if (!solutionEntity.TryGetComponent(out SolutionContainerComponent contents))
+                return;
+
+            var solution = contents.SplitSolution(contents.MaxVolume);
+            // We take the square root so it becomes harder to reach higher amount values
+            var amount = (int) Math.Round(_rangeConstant + _rangeMultiplier*Math.Sqrt(intensity));
+            amount = Math.Min(amount, _maxRange);
+
+            if (_diluteReagents)
+            {
+                // The maximum value of solutionFraction is _reagentMaxConcentrationFactor, achieved when amount = 0
+                // The infimum of solutionFraction is 0, which is approached when amount tends to infinity
+                // solutionFraction is equal to 1 only when amount equals _reagentDilutionStart
+                float solutionFraction;
+                if (amount >= _reagentDilutionStart)
+                {
+                    // Weird formulas here but basically when amount increases, solutionFraction gets closer to 0 in a reciprocal manner
+                    // _reagentDilutionFactor defines how fast solutionFraction gets closer to 0
+                    solutionFraction = 1 / (_reagentDilutionFactor*(amount - _reagentDilutionStart) + 1);
+                }
+                else
+                {
+                    // Here when amount decreases, solutionFraction gets closer to _reagentMaxConcentrationFactor in a linear manner
+                    solutionFraction = amount * (1 - _reagentMaxConcentrationFactor) / _reagentDilutionStart +
+                                       _reagentMaxConcentrationFactor;
+                }
+                solution.RemoveSolution(solution.TotalVolume*(1-solutionFraction));
+            }
+
+            if (!_mapManager.TryFindGridAt(solutionEntity.Transform.MapPosition, out var grid)) return;
+
+            var coords = grid.MapToGrid(solutionEntity.Transform.MapPosition);
+            var ent = solutionEntity.EntityManager.SpawnEntity(_prototypeId, coords.SnapToGrid());
```

Comment:
default value for _prototypeId is null, so this will just spawn an entity without any components if no prototype is provided :/ either check in ExposeData if a prototype was provided or handle it accordingly here